streams/src/test/java/org/apache/kafka/test/MockClientSupplier.java

```diff
@@ -58,7 +58,7 @@ public void setClusterForAdminClient(final Cluster cluster) {
 
     @Override
     public Admin getAdmin(final Map<String, Object> config) {
-        return new MockAdminClient(cluster.nodes(), cluster.nodeById(0));
+        return new MockAdminClient(cluster.nodes(), cluster.nodeById(-1));
```

Comment:
This is a piggy-backed fix: the controller's id is defined as -1, so with a single node cluster we should set it's controller as -1 not as 0.